internal/impl/packages.go

```diff
@@ -221,6 +235,17 @@ func (d *Devbox) ensurePackagesAreInstalled(ctx context.Context, mode installMod
 		return err
 	}
 
+	// Update lockfile with new packages that are not to be installed
+	cfgPkgs, err := d.ConfigPackages()
+	if err != nil {
+		return err
+	}
+	for _, pkg := range cfgPkgs {
+		if err := pkg.EnsureUninstallableIsInLockfile(); err != nil {
+			return err
+		}
+	}
```

Comment:
EnsureUninstallableIsInLockfile is a bit awkward, you could simply loop on `d.ConfigPackages()` and resolve all.

If they are already resolved, it's a no-op.

Reply:
it has some more checks inside the function....